{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5415357b-a033-43aa-b61e-0e15989339ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting python-dotenv\n",
      "  Downloading python_dotenv-1.0.0-py3-none-any.whl (19 kB)\n",
      "Installing collected packages: python-dotenv\n",
      "Successfully installed python-dotenv-1.0.0\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Requirement already satisfied: boto3 in /opt/app-root/lib/python3.8/site-packages (1.17.11)\n",
      "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in /opt/app-root/lib/python3.8/site-packages (from boto3) (0.3.7)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /opt/app-root/lib/python3.8/site-packages (from boto3) (0.10.0)\n",
      "Requirement already satisfied: botocore<1.21.0,>=1.20.11 in /opt/app-root/lib/python3.8/site-packages (from boto3) (1.20.101)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.25.4 in /opt/app-root/lib/python3.8/site-packages (from botocore<1.21.0,>=1.20.11->boto3) (1.26.6)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /opt/app-root/lib/python3.8/site-packages (from botocore<1.21.0,>=1.20.11->boto3) (2.8.1)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.21.0,>=1.20.11->boto3) (1.16.0)\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting trino\n",
      "  Downloading trino-0.326.0-py3-none-any.whl (44 kB)\n",
      "\u001b[K     |████████████████████████████████| 44 kB 29.7 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: sqlalchemy in /opt/app-root/lib/python3.8/site-packages (1.4.19)\n",
      "Collecting sqlalchemy-trino\n",
      "  Downloading sqlalchemy_trino-0.5.0-py3-none-any.whl (6.8 kB)\n",
      "Requirement already satisfied: python-dateutil in /opt/app-root/lib/python3.8/site-packages (from trino) (2.8.1)\n",
      "Requirement already satisfied: pytz in /opt/app-root/lib/python3.8/site-packages (from trino) (2021.1)\n",
      "Collecting backports.zoneinfo\n",
      "  Downloading backports.zoneinfo-0.2.1-cp38-cp38-manylinux1_x86_64.whl (74 kB)\n",
      "\u001b[K     |████████████████████████████████| 74 kB 103.0 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting requests>=2.31.0\n",
      "  Downloading requests-2.31.0-py3-none-any.whl (62 kB)\n",
      "\u001b[K     |████████████████████████████████| 62 kB 76.3 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: tzlocal in /opt/app-root/lib/python3.8/site-packages (from trino) (2.1)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino) (2021.5.30)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino) (2.10)\n",
      "Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino) (1.26.6)\n",
      "Collecting charset-normalizer<4,>=2\n",
      "  Downloading charset_normalizer-3.2.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (199 kB)\n",
      "\u001b[K     |████████████████████████████████| 199 kB 129.9 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: greenlet!=0.4.17 in /opt/app-root/lib/python3.8/site-packages (from sqlalchemy) (1.1.0)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil->trino) (1.16.0)\n",
      "Installing collected packages: charset-normalizer, requests, backports.zoneinfo, trino, sqlalchemy-trino\n",
      "  Attempting uninstall: requests\n",
      "    Found existing installation: requests 2.25.1\n",
      "    Uninstalling requests-2.25.1:\n",
      "      Successfully uninstalled requests-2.25.1\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "thoth-python 0.16.1 requires beautifulsoup4==4.6.3, but you have beautifulsoup4 4.9.3 which is incompatible.\u001b[0m\n",
      "Successfully installed backports.zoneinfo-0.2.1 charset-normalizer-3.2.0 requests-2.31.0 sqlalchemy-trino-0.5.0 trino-0.326.0\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Requirement already satisfied: pandas in /opt/app-root/lib/python3.8/site-packages (1.2.5)\n",
      "Requirement already satisfied: pyarrow in /opt/app-root/lib/python3.8/site-packages (4.0.1)\n",
      "Collecting fastparquet\n",
      "  Downloading fastparquet-2023.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.7 MB)\n",
      "\u001b[K     |████████████████████████████████| 1.7 MB 27.5 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pytz>=2017.3 in /opt/app-root/lib/python3.8/site-packages (from pandas) (2021.1)\n",
      "Requirement already satisfied: numpy>=1.16.5 in /opt/app-root/lib/python3.8/site-packages (from pandas) (1.21.0)\n",
      "Requirement already satisfied: python-dateutil>=2.7.3 in /opt/app-root/lib/python3.8/site-packages (from pandas) (2.8.1)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil>=2.7.3->pandas) (1.16.0)\n",
      "Requirement already satisfied: packaging in /opt/app-root/lib/python3.8/site-packages (from fastparquet) (20.9)\n",
      "Collecting cramjam>=2.3\n",
      "  Downloading cramjam-2.7.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)\n",
      "\u001b[K     |████████████████████████████████| 1.6 MB 134.7 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting pandas\n",
      "  Downloading pandas-2.0.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.4 MB)\n",
      "\u001b[K     |████████████████████████████████| 12.4 MB 127.4 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: fsspec in /opt/app-root/lib/python3.8/site-packages (from fastparquet) (2021.6.1)\n",
      "Collecting tzdata>=2022.1\n",
      "  Downloading tzdata-2023.3-py2.py3-none-any.whl (341 kB)\n",
      "\u001b[K     |████████████████████████████████| 341 kB 130.9 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting python-dateutil>=2.7.3\n",
      "  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
      "\u001b[K     |████████████████████████████████| 247 kB 129.2 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pyparsing>=2.0.2 in /opt/app-root/lib/python3.8/site-packages (from packaging->fastparquet) (2.4.7)\n",
      "Installing collected packages: tzdata, python-dateutil, pandas, cramjam, fastparquet\n",
      "  Attempting uninstall: python-dateutil\n",
      "    Found existing installation: python-dateutil 2.8.1\n",
      "    Uninstalling python-dateutil-2.8.1:\n",
      "      Successfully uninstalled python-dateutil-2.8.1\n",
      "  Attempting uninstall: pandas\n",
      "    Found existing installation: pandas 1.2.5\n",
      "    Uninstalling pandas-1.2.5:\n",
      "      Successfully uninstalled pandas-1.2.5\n",
      "Successfully installed cramjam-2.7.0 fastparquet-2023.8.0 pandas-2.0.3 python-dateutil-2.8.2 tzdata-2023.3\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting osc-ingest-tools\n",
      "  Downloading osc_ingest_tools-0.4.3-py3-none-any.whl (14 kB)\n",
      "Requirement already satisfied: trino[sqlalchemy] in /opt/app-root/lib/python3.8/site-packages (from osc-ingest-tools) (0.326.0)\n",
      "Requirement already satisfied: boto3 in /opt/app-root/lib/python3.8/site-packages (from osc-ingest-tools) (1.17.11)\n",
      "Requirement already satisfied: pandas in /opt/app-root/lib/python3.8/site-packages (from osc-ingest-tools) (2.0.3)\n",
      "Requirement already satisfied: python-dotenv in /opt/app-root/lib/python3.8/site-packages (from osc-ingest-tools) (1.0.0)\n",
      "Requirement already satisfied: sqlalchemy in /opt/app-root/lib/python3.8/site-packages (from osc-ingest-tools) (1.4.19)\n",
      "Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in /opt/app-root/lib/python3.8/site-packages (from boto3->osc-ingest-tools) (0.10.0)\n",
      "Requirement already satisfied: s3transfer<0.4.0,>=0.3.0 in /opt/app-root/lib/python3.8/site-packages (from boto3->osc-ingest-tools) (0.3.7)\n",
      "Requirement already satisfied: botocore<1.21.0,>=1.20.11 in /opt/app-root/lib/python3.8/site-packages (from boto3->osc-ingest-tools) (1.20.101)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.25.4 in /opt/app-root/lib/python3.8/site-packages (from botocore<1.21.0,>=1.20.11->boto3->osc-ingest-tools) (1.26.6)\n",
      "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /opt/app-root/lib/python3.8/site-packages (from botocore<1.21.0,>=1.20.11->boto3->osc-ingest-tools) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.21.0,>=1.20.11->boto3->osc-ingest-tools) (1.16.0)\n",
      "Requirement already satisfied: numpy>=1.20.3 in /opt/app-root/lib/python3.8/site-packages (from pandas->osc-ingest-tools) (1.21.0)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /opt/app-root/lib/python3.8/site-packages (from pandas->osc-ingest-tools) (2023.3)\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/app-root/lib/python3.8/site-packages (from pandas->osc-ingest-tools) (2021.1)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in /opt/app-root/lib/python3.8/site-packages (from sqlalchemy->osc-ingest-tools) (1.1.0)\n",
      "Requirement already satisfied: tzlocal in /opt/app-root/lib/python3.8/site-packages (from trino[sqlalchemy]->osc-ingest-tools) (2.1)\n",
      "Requirement already satisfied: backports.zoneinfo in /opt/app-root/lib/python3.8/site-packages (from trino[sqlalchemy]->osc-ingest-tools) (0.2.1)\n",
      "Requirement already satisfied: requests>=2.31.0 in /opt/app-root/lib/python3.8/site-packages (from trino[sqlalchemy]->osc-ingest-tools) (2.31.0)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino[sqlalchemy]->osc-ingest-tools) (3.2.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino[sqlalchemy]->osc-ingest-tools) (2021.5.30)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /opt/app-root/lib/python3.8/site-packages (from requests>=2.31.0->trino[sqlalchemy]->osc-ingest-tools) (2.10)\n",
      "Installing collected packages: osc-ingest-tools\n",
      "Successfully installed osc-ingest-tools-0.4.3\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Requirement already satisfied: ipython in /opt/app-root/lib/python3.8/site-packages (7.25.0)\n",
      "Collecting ipython\n",
      "  Downloading ipython-8.12.2-py3-none-any.whl (797 kB)\n",
      "\u001b[K     |████████████████████████████████| 797 kB 25.5 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pygments>=2.4.0 in /opt/app-root/lib/python3.8/site-packages (from ipython) (2.9.0)\n",
      "Requirement already satisfied: decorator in /opt/app-root/lib/python3.8/site-packages (from ipython) (4.4.2)\n",
      "Requirement already satisfied: pickleshare in /opt/app-root/lib/python3.8/site-packages (from ipython) (0.7.5)\n",
      "Requirement already satisfied: pexpect>4.3 in /opt/app-root/lib/python3.8/site-packages (from ipython) (4.8.0)\n",
      "Collecting prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30\n",
      "  Downloading prompt_toolkit-3.0.39-py3-none-any.whl (385 kB)\n",
      "\u001b[K     |████████████████████████████████| 385 kB 136.7 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: jedi>=0.16 in /opt/app-root/lib/python3.8/site-packages (from ipython) (0.17.2)\n",
      "Requirement already satisfied: backcall in /opt/app-root/lib/python3.8/site-packages (from ipython) (0.2.0)\n",
      "Collecting stack-data\n",
      "  Downloading stack_data-0.6.2-py3-none-any.whl (24 kB)\n",
      "Requirement already satisfied: traitlets>=5 in /opt/app-root/lib/python3.8/site-packages (from ipython) (5.0.5)\n",
      "Requirement already satisfied: typing-extensions in /opt/app-root/lib/python3.8/site-packages (from ipython) (3.10.0.0)\n",
      "Requirement already satisfied: matplotlib-inline in /opt/app-root/lib/python3.8/site-packages (from ipython) (0.1.2)\n",
      "Requirement already satisfied: parso<0.8.0,>=0.7.0 in /opt/app-root/lib/python3.8/site-packages (from jedi>=0.16->ipython) (0.7.1)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /opt/app-root/lib/python3.8/site-packages (from pexpect>4.3->ipython) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in /opt/app-root/lib/python3.8/site-packages (from prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30->ipython) (0.2.5)\n",
      "Requirement already satisfied: ipython-genutils in /opt/app-root/lib/python3.8/site-packages (from traitlets>=5->ipython) (0.2.0)\n",
      "Collecting executing>=1.2.0\n",
      "  Downloading executing-1.2.0-py2.py3-none-any.whl (24 kB)\n",
      "Collecting pure-eval\n",
      "  Downloading pure_eval-0.2.2-py3-none-any.whl (11 kB)\n",
      "Collecting asttokens>=2.1.0\n",
      "  Downloading asttokens-2.4.0-py2.py3-none-any.whl (27 kB)\n",
      "Requirement already satisfied: six>=1.12.0 in /opt/app-root/lib/python3.8/site-packages (from asttokens>=2.1.0->stack-data->ipython) (1.16.0)\n",
      "Installing collected packages: pure-eval, executing, asttokens, stack-data, prompt-toolkit, ipython\n",
      "  Attempting uninstall: prompt-toolkit\n",
      "    Found existing installation: prompt-toolkit 3.0.19\n",
      "    Uninstalling prompt-toolkit-3.0.19:\n",
      "      Successfully uninstalled prompt-toolkit-3.0.19\n",
      "  Attempting uninstall: ipython\n",
      "    Found existing installation: ipython 7.25.0\n",
      "    Uninstalling ipython-7.25.0:\n",
      "      Successfully uninstalled ipython-7.25.0\n",
      "Successfully installed asttokens-2.4.0 executing-1.2.0 ipython-8.12.2 prompt-toolkit-3.0.39 pure-eval-0.2.2 stack-data-0.6.2\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting country_converter\n",
      "  Downloading country_converter-1.0.0-py3-none-any.whl (44 kB)\n",
      "\u001b[K     |████████████████████████████████| 44 kB 25.0 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pandas>=1.0 in /opt/app-root/lib/python3.8/site-packages (from country_converter) (2.0.3)\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.0->country_converter) (2021.1)\n",
      "Requirement already satisfied: numpy>=1.20.3 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.0->country_converter) (1.21.0)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.0->country_converter) (2023.3)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.0->country_converter) (2.8.2)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil>=2.8.2->pandas>=1.0->country_converter) (1.16.0)\n",
      "Installing collected packages: country-converter\n",
      "Successfully installed country-converter-1.0.0\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting pint-pandas\n",
      "  Downloading Pint_Pandas-0.3-py2.py3-none-any.whl (23 kB)\n",
      "Requirement already satisfied: pandas>=1.5 in /opt/app-root/lib/python3.8/site-packages (from pint-pandas) (2.0.3)\n",
      "Collecting pint>=0.20.1\n",
      "  Downloading Pint-0.21.1-py3-none-any.whl (290 kB)\n",
      "\u001b[K     |████████████████████████████████| 290 kB 47.0 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: numpy>=1.20.3 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.5->pint-pandas) (1.21.0)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.5->pint-pandas) (2023.3)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.5->pint-pandas) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/app-root/lib/python3.8/site-packages (from pandas>=1.5->pint-pandas) (2021.1)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil>=2.8.2->pandas>=1.5->pint-pandas) (1.16.0)\n",
      "Installing collected packages: pint, pint-pandas\n",
      "Successfully installed pint-0.21.1 pint-pandas-0.3\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting openscm-units\n",
      "  Downloading openscm_units-0.5.2-py2.py3-none-any.whl (13 kB)\n",
      "Requirement already satisfied: pandas in /opt/app-root/lib/python3.8/site-packages (from openscm-units) (2.0.3)\n",
      "Collecting globalwarmingpotentials\n",
      "  Downloading globalwarmingpotentials-0.9.3-py2.py3-none-any.whl (12 kB)\n",
      "Requirement already satisfied: pint in /opt/app-root/lib/python3.8/site-packages (from openscm-units) (0.21.1)\n",
      "Requirement already satisfied: numpy>=1.20.3 in /opt/app-root/lib/python3.8/site-packages (from pandas->openscm-units) (1.21.0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/app-root/lib/python3.8/site-packages (from pandas->openscm-units) (2021.1)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /opt/app-root/lib/python3.8/site-packages (from pandas->openscm-units) (2.8.2)\n",
      "Requirement already satisfied: tzdata>=2022.1 in /opt/app-root/lib/python3.8/site-packages (from pandas->openscm-units) (2023.3)\n",
      "Requirement already satisfied: six>=1.5 in /opt/app-root/lib/python3.8/site-packages (from python-dateutil>=2.8.2->pandas->openscm-units) (1.16.0)\n",
      "Installing collected packages: globalwarmingpotentials, openscm-units\n",
      "Successfully installed globalwarmingpotentials-0.9.3 openscm-units-0.5.2\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting Pint==0.19.2\n",
      "  Downloading Pint-0.19.2.tar.gz (292 kB)\n",
      "\u001b[K     |████████████████████████████████| 292 kB 26.8 MB/s eta 0:00:01\n",
      "\u001b[?25h  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h    Preparing wheel metadata ... \u001b[?25ldone\n",
      "\u001b[?25hBuilding wheels for collected packages: Pint\n",
      "  Building wheel for Pint (PEP 517) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for Pint: filename=Pint-0.19.2-py3-none-any.whl size=230998 sha256=bdb32c220a4c18e463eff17c88f28e93fd61f994bd5842630ddaad6c6ac99055\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-z5yee2ly/wheels/98/1d/e0/a44541f5302ac50028a3be0db9417c18092c61f25009d62145\n",
      "Successfully built Pint\n",
      "Installing collected packages: Pint\n",
      "  Attempting uninstall: Pint\n",
      "    Found existing installation: Pint 0.21.1\n",
      "    Uninstalling Pint-0.21.1:\n",
      "      Successfully uninstalled Pint-0.21.1\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "pint-pandas 0.3 requires pint>=0.20.1, but you have pint 0.19.2 which is incompatible.\u001b[0m\n",
      "Successfully installed Pint-0.19.2\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Requirement already satisfied: jedi in /opt/app-root/lib/python3.8/site-packages (0.17.2)\n",
      "Requirement already satisfied: parso<0.8.0,>=0.7.0 in /opt/app-root/lib/python3.8/site-packages (from jedi) (0.7.1)\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting pycountry\n",
      "  Downloading pycountry-22.3.5.tar.gz (10.1 MB)\n",
      "\u001b[K     |████████████████████████████████| 10.1 MB 27.3 MB/s eta 0:00:01\n",
      "\u001b[?25h  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h    Preparing wheel metadata ... \u001b[?25ldone\n",
      "\u001b[?25hRequirement already satisfied: setuptools in /opt/app-root/lib/python3.8/site-packages (from pycountry) (57.0.0)\n",
      "Building wheels for collected packages: pycountry\n",
      "  Building wheel for pycountry (PEP 517) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for pycountry: filename=pycountry-22.3.5-py2.py3-none-any.whl size=10681833 sha256=67aaa37fcfc7d39627a2f7d81d77d26fa62b8f24a65ad980bd27e839c82663a5\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-dk8kpnpf/wheels/e2/aa/0f/c224e473b464387170b83ca7c66947b4a7e33e8d903a679748\n",
      "Successfully built pycountry\n",
      "Installing collected packages: pycountry\n",
      "Successfully installed pycountry-22.3.5\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting openpyxl\n",
      "  Downloading openpyxl-3.1.2-py2.py3-none-any.whl (249 kB)\n",
      "\u001b[K     |████████████████████████████████| 249 kB 25.7 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting et-xmlfile\n",
      "  Downloading et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)\n",
      "Installing collected packages: et-xmlfile, openpyxl\n",
      "Successfully installed et-xmlfile-1.1.0 openpyxl-3.1.2\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting xlwt\n",
      "  Downloading xlwt-1.3.0-py2.py3-none-any.whl (99 kB)\n",
      "\u001b[K     |████████████████████████████████| 99 kB 24.1 MB/s eta 0:00:01\n",
      "\u001b[?25hInstalling collected packages: xlwt\n",
      "Successfully installed xlwt-1.3.0\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n",
      "Collecting dbt-core\n",
      "  Downloading dbt_core-1.6.2-py3-none-any.whl (992 kB)\n",
      "\u001b[K     |████████████████████████████████| 992 kB 24.2 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting dbt-trino\n",
      "  Downloading dbt_trino-1.6.1-py3-none-any.whl (32 kB)\n",
      "Requirement already satisfied: cffi<2.0.0,>=1.9 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (1.14.5)\n",
      "Collecting logbook<1.6,>=1.5\n",
      "  Downloading Logbook-1.5.3.tar.gz (85 kB)\n",
      "\u001b[K     |████████████████████████████████| 85 kB 111.5 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting minimal-snowplow-tracker~=0.0.2\n",
      "  Downloading minimal-snowplow-tracker-0.0.2.tar.gz (12 kB)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (3.10.0.0)\n",
      "Collecting protobuf>=4.0.0\n",
      "  Downloading protobuf-4.24.3-cp37-abi3-manylinux2014_x86_64.whl (311 kB)\n",
      "\u001b[K     |████████████████████████████████| 311 kB 115.4 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting Jinja2~=3.1.2\n",
      "  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)\n",
      "\u001b[K     |████████████████████████████████| 133 kB 132.7 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: click<9 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (8.0.1)\n",
      "Collecting agate~=1.7.0\n",
      "  Downloading agate-1.7.1-py2.py3-none-any.whl (97 kB)\n",
      "\u001b[K     |████████████████████████████████| 97 kB 129.3 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting hologram~=0.0.16\n",
      "  Downloading hologram-0.0.16-py3-none-any.whl (11 kB)\n",
      "Requirement already satisfied: networkx<4,>=2.3 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (2.5.1)\n",
      "Requirement already satisfied: colorama<0.5,>=0.3.9 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (0.4.4)\n",
      "Collecting dbt-semantic-interfaces~=0.2.0\n",
      "  Downloading dbt_semantic_interfaces-0.2.0-py3-none-any.whl (99 kB)\n",
      "\u001b[K     |████████████████████████████████| 99 kB 126.2 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting mashumaro[msgpack]~=3.8.1\n",
      "  Downloading mashumaro-3.8.1-py3-none-any.whl (74 kB)\n",
      "\u001b[K     |████████████████████████████████| 74 kB 104.7 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: idna<4,>=2.5 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (2.10)\n",
      "Collecting isodate<0.7,>=0.6\n",
      "  Downloading isodate-0.6.1-py2.py3-none-any.whl (41 kB)\n",
      "\u001b[K     |████████████████████████████████| 41 kB 41.8 MB/s  eta 0:00:01\n",
      "\u001b[?25hCollecting dbt-extractor~=0.4.1\n",
      "  Downloading dbt_extractor-0.4.1-cp36-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
      "\u001b[K     |████████████████████████████████| 1.1 MB 127.1 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting pathspec<0.12,>=0.9\n",
      "  Downloading pathspec-0.11.2-py3-none-any.whl (29 kB)\n",
      "Requirement already satisfied: requests<3.0.0 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (2.31.0)\n",
      "Requirement already satisfied: urllib3~=1.0 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (1.26.6)\n",
      "Collecting sqlparse<0.5,>=0.2.3\n",
      "  Downloading sqlparse-0.4.4-py3-none-any.whl (41 kB)\n",
      "\u001b[K     |████████████████████████████████| 41 kB 17.3 MB/s  eta 0:00:01\n",
      "\u001b[?25hCollecting packaging>20.9\n",
      "  Downloading packaging-23.1-py3-none-any.whl (48 kB)\n",
      "\u001b[K     |████████████████████████████████| 48 kB 106.4 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pytz>=2015.7 in /opt/app-root/lib/python3.8/site-packages (from dbt-core) (2021.1)\n",
      "Collecting pyyaml>=6.0\n",
      "  Downloading PyYAML-6.0.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (736 kB)\n",
      "\u001b[K     |████████████████████████████████| 736 kB 120.6 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting python-slugify>=1.2.1\n",
      "  Downloading python_slugify-8.0.1-py2.py3-none-any.whl (9.7 kB)\n",
      "Requirement already satisfied: Babel>=2.0 in /opt/app-root/lib/python3.8/site-packages (from agate~=1.7.0->dbt-core) (2.9.1)\n",
      "Collecting parsedatetime!=2.5,>=2.1\n",
      "  Downloading parsedatetime-2.6-py3-none-any.whl (42 kB)\n",
      "\u001b[K     |████████████████████████████████| 42 kB 59.4 MB/s  eta 0:00:01\n",
      "\u001b[?25hCollecting pytimeparse>=1.1.5\n",
      "  Downloading pytimeparse-1.1.8-py2.py3-none-any.whl (10.0 kB)\n",
      "Collecting leather>=0.3.2\n",
      "  Downloading leather-0.3.4-py2.py3-none-any.whl (31 kB)\n",
      "Requirement already satisfied: pycparser in /opt/app-root/lib/python3.8/site-packages (from cffi<2.0.0,>=1.9->dbt-core) (2.20)\n",
      "Collecting typing-extensions>=3.7.4\n",
      "  Downloading typing_extensions-4.7.1-py3-none-any.whl (33 kB)\n",
      "Collecting more-itertools~=8.0\n",
      "  Downloading more_itertools-8.14.0-py3-none-any.whl (52 kB)\n",
      "\u001b[K     |████████████████████████████████| 52 kB 61.2 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: python-dateutil~=2.0 in /opt/app-root/lib/python3.8/site-packages (from dbt-semantic-interfaces~=0.2.0->dbt-core) (2.8.2)\n",
      "Requirement already satisfied: jsonschema~=3.0 in /opt/app-root/lib/python3.8/site-packages (from dbt-semantic-interfaces~=0.2.0->dbt-core) (3.2.0)\n",
      "Collecting importlib-metadata~=6.0\n",
      "  Downloading importlib_metadata-6.8.0-py3-none-any.whl (22 kB)\n",
      "Collecting pydantic~=1.10\n",
      "  Downloading pydantic-1.10.12-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)\n",
      "\u001b[K     |████████████████████████████████| 3.2 MB 117.8 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: zipp>=0.5 in /opt/app-root/lib/python3.8/site-packages (from importlib-metadata~=6.0->dbt-semantic-interfaces~=0.2.0->dbt-core) (3.4.1)\n",
      "Requirement already satisfied: six in /opt/app-root/lib/python3.8/site-packages (from isodate<0.7,>=0.6->dbt-core) (1.16.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in /opt/app-root/lib/python3.8/site-packages (from Jinja2~=3.1.2->dbt-core) (2.0.1)\n",
      "Requirement already satisfied: pyrsistent>=0.14.0 in /opt/app-root/lib/python3.8/site-packages (from jsonschema~=3.0->dbt-semantic-interfaces~=0.2.0->dbt-core) (0.17.3)\n",
      "Requirement already satisfied: setuptools in /opt/app-root/lib/python3.8/site-packages (from jsonschema~=3.0->dbt-semantic-interfaces~=0.2.0->dbt-core) (57.0.0)\n",
      "Requirement already satisfied: attrs>=17.4.0 in /opt/app-root/lib/python3.8/site-packages (from jsonschema~=3.0->dbt-semantic-interfaces~=0.2.0->dbt-core) (21.2.0)\n",
      "Requirement already satisfied: msgpack>=0.5.6 in /opt/app-root/lib/python3.8/site-packages (from mashumaro[msgpack]~=3.8.1->dbt-core) (1.0.2)\n",
      "Requirement already satisfied: decorator<5,>=4.3 in /opt/app-root/lib/python3.8/site-packages (from networkx<4,>=2.3->dbt-core) (4.4.2)\n",
      "Collecting text-unidecode>=1.3\n",
      "  Downloading text_unidecode-1.3-py2.py3-none-any.whl (78 kB)\n",
      "\u001b[K     |████████████████████████████████| 78 kB 118.6 MB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: charset-normalizer<4,>=2 in /opt/app-root/lib/python3.8/site-packages (from requests<3.0.0->dbt-core) (3.2.0)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /opt/app-root/lib/python3.8/site-packages (from requests<3.0.0->dbt-core) (2021.5.30)\n",
      "Requirement already satisfied: trino~=0.326 in /opt/app-root/lib/python3.8/site-packages (from dbt-trino) (0.326.0)\n",
      "Requirement already satisfied: tzlocal in /opt/app-root/lib/python3.8/site-packages (from trino~=0.326->dbt-trino) (2.1)\n",
      "Requirement already satisfied: backports.zoneinfo in /opt/app-root/lib/python3.8/site-packages (from trino~=0.326->dbt-trino) (0.2.1)\n",
      "Building wheels for collected packages: logbook, minimal-snowplow-tracker\n",
      "  Building wheel for logbook (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for logbook: filename=Logbook-1.5.3-cp38-cp38-linux_x86_64.whl size=66394 sha256=6fd2746aed7ffa1a09b80dfbb0b20f3bf5aef8f7cbd0d95470bfb22be4e6ffa7\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-1c2pkjfm/wheels/25/9a/c0/03bf368ac5f673e9c0bde58837146b004424b25555ca3aedfb\n",
      "  Building wheel for minimal-snowplow-tracker (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for minimal-snowplow-tracker: filename=minimal_snowplow_tracker-0.0.2-py3-none-any.whl size=17816 sha256=b80d99ce6fb5e7f744537840a14c137d4030ddc005887a3c6d5cec0030781a08\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-1c2pkjfm/wheels/d4/04/77/72a8c17c9c19691be497bb4e192bd760aff6d47fa959d184e8\n",
      "Successfully built logbook minimal-snowplow-tracker\n",
      "Installing collected packages: typing-extensions, text-unidecode, pyyaml, pytimeparse, python-slugify, pydantic, parsedatetime, more-itertools, mashumaro, leather, Jinja2, isodate, importlib-metadata, sqlparse, protobuf, pathspec, packaging, minimal-snowplow-tracker, logbook, hologram, dbt-semantic-interfaces, dbt-extractor, agate, dbt-core, dbt-trino\n",
      "  Attempting uninstall: typing-extensions\n",
      "    Found existing installation: typing-extensions 3.10.0.0\n",
      "    Uninstalling typing-extensions-3.10.0.0:\n",
      "      Successfully uninstalled typing-extensions-3.10.0.0\n",
      "  Attempting uninstall: pyyaml\n",
      "    Found existing installation: PyYAML 5.4.1\n",
      "    Uninstalling PyYAML-5.4.1:\n",
      "      Successfully uninstalled PyYAML-5.4.1\n",
      "  Attempting uninstall: Jinja2\n",
      "    Found existing installation: Jinja2 2.11.3\n",
      "    Uninstalling Jinja2-2.11.3:\n",
      "      Successfully uninstalled Jinja2-2.11.3\n",
      "  Attempting uninstall: importlib-metadata\n",
      "    Found existing installation: importlib-metadata 4.6.0\n",
      "    Uninstalling importlib-metadata-4.6.0:\n",
      "      Successfully uninstalled importlib-metadata-4.6.0\n",
      "  Attempting uninstall: protobuf\n",
      "    Found existing installation: protobuf 3.17.3\n",
      "    Uninstalling protobuf-3.17.3:\n",
      "      Successfully uninstalled protobuf-3.17.3\n",
      "  Attempting uninstall: pathspec\n",
      "    Found existing installation: pathspec 0.8.1\n",
      "    Uninstalling pathspec-0.8.1:\n",
      "      Successfully uninstalled pathspec-0.8.1\n",
      "  Attempting uninstall: packaging\n",
      "    Found existing installation: packaging 20.9\n",
      "    Uninstalling packaging-20.9:\n",
      "      Successfully uninstalled packaging-20.9\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
      "thoth-python 0.16.1 requires beautifulsoup4==4.6.3, but you have beautifulsoup4 4.9.3 which is incompatible.\n",
      "kfp-pipeline-spec 0.1.8 requires protobuf<4,>=3.13.0, but you have protobuf 4.24.3 which is incompatible.\n",
      "elyra-server 2.2.4 requires jinja2<3.0,>=2.11, but you have jinja2 3.1.2 which is incompatible.\n",
      "elyra-server 2.2.4 requires pyyaml<6.0,>=5.3.1, but you have pyyaml 6.0.1 which is incompatible.\u001b[0m\n",
      "Successfully installed Jinja2-3.1.2 agate-1.7.1 dbt-core-1.6.2 dbt-extractor-0.4.1 dbt-semantic-interfaces-0.2.0 dbt-trino-1.6.1 hologram-0.0.16 importlib-metadata-6.8.0 isodate-0.6.1 leather-0.3.4 logbook-1.5.3 mashumaro-3.8.1 minimal-snowplow-tracker-0.0.2 more-itertools-8.14.0 packaging-23.1 parsedatetime-2.6 pathspec-0.11.2 protobuf-4.24.3 pydantic-1.10.12 python-slugify-8.0.1 pytimeparse-1.1.8 pyyaml-6.0.1 sqlparse-0.4.4 text-unidecode-1.3 typing-extensions-4.7.1\n",
      "\u001b[33mWARNING: You are using pip version 21.1.3; however, version 23.2.1 is available.\n",
      "You should consider upgrading via the '/opt/app-root/bin/python3.8 -m pip install --upgrade pip' command.\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "# 'capture' magic prevents long outputs from spamming your notebook\n",
    "#%%capture pipoutput\n",
    "\n",
    "# For loading predefined environment variables from files\n",
    "# Typically used to load sensitive access credentials\n",
    "%pip install python-dotenv\n",
    "\n",
    "# Standard python package for interacting with S3 buckets\n",
    "%pip install boto3\n",
    "# Interacting with Trino and using Trino with sqlalchemy\n",
    "%pip install trino sqlalchemy sqlalchemy-trino\n",
    "\n",
    "# Pandas and parquet file i/o\n",
    "%pip install pandas pyarrow fastparquet\n",
    "\n",
    "# OS-Climate utilities to make data ingest easier\n",
    "%pip install osc-ingest-tools\n",
    "%pip install ipython --upgrade\n",
    "%pip install country_converter --upgrade\n",
    "%pip install pint-pandas\n",
    "%pip install openscm-units\n",
    "%pip install Pint==0.19.2\n",
    "%pip install jedi\n",
    "%pip install pycountry\n",
    "%pip install openpyxl\n",
    "%pip install xlwt\n",
    "%pip install dbt-core dbt-trino"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "548a2b9e-1041-4708-b25b-31a0b251ced0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c84a9a0-6098-40ef-8a89-d2adcc22baae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a5d5213-4849-4f21-a1fc-f86b7196aae3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
